File: Leetcode/2442. Count Number of Distinct Integers After Reverse Operations/solution.go
func countDistinctIntegers(nums []int) int {
	distinctNums := make(map[int]bool)
	for _, num := range nums {
		distinctNums[num] = true
		reversedNum, _ := strconv.Atoi(reverseNumber(num))
		distinctNums[reversedNum] = true
	}
	return len(distinctNums)
}

func reverseNumber(num int) string {
	numStr := strconv.Itoa(num)
	reversed := ""
	for i := len(numStr) - 1; i >= 0; i-- {
		reversed += string(numStr[i])
	}
	return reversed
}

File: Leetcode/2442. Count Number of Distinct Integers After Reverse Operations/solution.rs
use std::collections::HashSet;

impl Solution {
    pub fn count_distinct_integers(nums: Vec<i32>) -> i32 {
        let mut distinct_nums = HashSet::new();
        for &num in &nums {
            distinct_nums.insert(num);
            distinct_nums.insert(num.to_string().chars().rev().collect::<String>().parse().unwrap());
        }
        distinct_nums.len() as i32
    }
}